### PR TITLE
Add a PostgreSQL database

### PR DESCRIPTION
**Pull request by felixclack**
We have decided to use PostgreSQL as our data store.

See [this ADR](https://github.com/DFE-Digital/find-a-lost-trn/blob/main/adr/00004-use-postgres-state.md) for the context on that decision.

This sets up PostgreSQL locally, in Docker and also on PaaS.

### Checklist

- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
 # Do not set this db to the same as development or production.
 test:
   <<: *default
-  database: db/test.sqlite3
+  database: find_my_trn_test
 
 production:
   <<: *default
-  database: db/production.sqlite3
+  database: find_my_trn_production
```
Comment (dankmitchell) [inline config/database.yml:21]:
```suggestion
  database: find_a_lost_trn_production
```

File: config/database.yml [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
-  database: db/development.sqlite3
+  database: find_my_trn_development
 
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
 # Do not set this db to the same as development or production.
 test:
   <<: *default
-  database: db/test.sqlite3
+  database: find_my_trn_test
```
Comment (dankmitchell) [inline config/database.yml:17]:
```suggestion
  database: find_a_lost_trn_test
```

File: config/database.yml [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
+  encoding: unicode
+  password: <%= ENV.fetch('DB_PASSWORD', '') %>
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-  timeout: 5000
+  username: <%= ENV.fetch('DB_USERNAME', 'postgres') %>
 
 development:
   <<: *default
-  database: db/development.sqlite3
+  database: find_my_trn_development
```
Comment (dankmitchell) [inline config/database.yml:10]:
```suggestion
  database: find_a_lost_trn_development
```

**Review by dankmitchell — COMMENTED** [review]
Just some updates to the name change

File: docker-compose.yml [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+    restart: always
   web:
     build: .
     command: /bin/sh -c "rm -f tmp/pids/server.pid && bundle exec rails server -p 3000 -b '0.0.0.0'"
     depends_on:
       - css
+      - db
       - js
+    environment:
+      - DB_DATABASE=find_my_trn_development
```
[the remark this answers is not in the document]
Comment (felixclack) [reply docker-compose.yml:22]: Good catch.

**Review by tvararu — APPROVED** [review]

**Pull request merged**